src/PAModel/MsAppTest.cs

```diff
@@ -86,85 +86,66 @@ public static bool Compare(string pathToZip1, string pathToZip2, TextWriter log)
             // If there's a checksum mismatch, do a more intensive comparison to find the difference.
 #if DEBUG
             // Provide a comparison that can be very specific about what the difference is.
-            Dictionary<string, string> comp = new Dictionary<string, string>();
-            var h1 = Test(pathToZip1, log, comp, true);
-            var h2 = Test(pathToZip2, log, comp, false);
-
+            var comp = new Dictionary<string, byte[]>();
+            DebugChecksum(pathToZip1, log, comp, true);
+            DebugChecksum(pathToZip2, log, comp, false);
 
             foreach (var kv in comp) // Remaining entries are errors.
             {
                 Console.WriteLine("FAIL: 2nd is missing " + kv.Key);
             }
-
-            if (h1 == h2)
-            {
-                log.WriteLine("Same!");
-                return true;
-            }
-            Console.WriteLine("FAIL!!");
 #endif
             return false;
         }
 
+        // Compare the debug checksums. 
         // Get a hash for the MsApp file.
         // First pass adds file/hash to comp.
         // Second pass checks hash equality and removes files from comp.
         // AFter second pass, comp should be 0. any files in comp were missing from 2nd pass.
-        public static string Test(string pathToZip, TextWriter log, Dictionary<string,string> comp, bool first)
+        public static void DebugChecksum(string pathToZip, TextWriter log, Dictionary<string,byte[]> comp, bool first)
```

Comment:
The real value of the change is making this debug function behave *exactly* the same as the checksum function. It shares the same json normalizing, double-escaping logic, etc. 
This runs the same checksum algorithm, but with a json writer (that captures the full output) rather than a sha256 incremental hash.